materialize-s3-parquet/s3_util.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+)
+
+// Uploader is the interface of the object responsible for uploading local files to the cloud.
+type Uploader interface {
+	Upload(bucket, key, localFileName string) error
+}
+
+// S3Uploader implements the Uploader interface for uploading files to S3.
+type S3Uploader struct {
+	uploaderImp *s3manager.Uploader
+}
+
+// Upload implements the S3Uploader interface.
+func (u *S3Uploader) Upload(bucket, key, localFileName string) error {
+	f, err := os.Open(localFileName)
+	if err != nil {
+		return fmt.Errorf("opening local file: %w", err)
+	}
+	defer f.Close()
+
+	upParams := &s3manager.UploadInput{
```

Comment:
I've added the logic that sets `contentType` to be general bin file of `binary/octet-stream`. Not sure if this is fine? (Or is there any other functions that requires more specific contentType?)

Reply:
huh, I would have thought that there was a specific mime type for parquet already, but after a quick google search it does not appear that such a mime type exists. One minor nit pick is that I think the mime type should be `application/octet-stream` instead of `binary/`, but apart from that it makes sense.